src/manager/InMemoryTaskManager.java

```diff
@@ -179,23 +195,46 @@ public void deleteTaskById(int idOfTask) {
 
     private void deduceEpicStatus(int idOfEpic) {
         List<Integer> listIdOfSubtasks = epicList.get(idOfEpic).getListIdOfSubtasks();
-        int countOfDone = 0;
-        if (listIdOfSubtasks.isEmpty()) epicList.get(idOfEpic).setStatus(Status.NEW);
-        for (int idSub : listIdOfSubtasks) {
-            if (subtaskList.get(idSub).getStatus() == Status.IN_PROGRESS) {
-                epicList.get(idOfEpic).setStatus(Status.IN_PROGRESS);
-                return;
-            } else if (subtaskList.get(idSub).getStatus() == Status.DONE) {
-                countOfDone++;
-            }
+        if (listIdOfSubtasks.isEmpty()) {
+            epicList.get(idOfEpic).setStatus(Status.NEW);
+            epicList.get(idOfEpic).setTimeFields(NO_TIME, NO_TIME, Duration.ofMinutes(0));
+            return;
         }
-        if (countOfDone == listIdOfSubtasks.size()) {
+        boolean allSubNew = listIdOfSubtasks.stream().map(idSub -> subtaskList.get(idSub).getStatus())
```

Comment:
Стримы принято разбивать по строкам для лучшей читаемости:
```suggestion
        boolean allSubNew = listIdOfSubtasks.stream()
            .map(idSub -> subtaskList.get(idSub).getStatus())
            .allMatch(status -> status == Status.NEW);
```
Поправь, пожалуйста, для всех стримов

Reply:
ok